src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -392,23 +395,148 @@ def list(self, nqn: str):
                 NVMeoFClient.pb2.list_connections_req(subsystem=nqn)
             )
 
+    @UIRouter('/nvmeof', Scope.NVME_OF)
+    class NVMeoFTcpUI(BaseController):
+        @Endpoint('GET', '/status')
+        @ReadPermission
+        @EndpointDoc("Display NVMe/TCP service Status",
+                     responses={200: NVME_SCHEMA})
+        def status(self) -> dict:
+            status: Dict[str, Any] = {'available': True, 'message': None}
+            orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+            if orch_backend == 'cephadm':
+                orch = OrchClient.instance()
+                orch_status = orch.status()
+                if not orch_status['available']:
+                    return status
+                if not orch.services.list_daemons(daemon_type='nvmeof'):
+                    status["available"] = False
+                    status["message"] = 'Create an NVMe/TCP service to get started.'
+            return status
+
+        @Endpoint('GET', '/subsystems')
+        @ReadPermission
+        def list(self):
+            response = []
+            # Get list of subystems
+            subsystems = NVMeoFClient().stub.list_subsystems(
+                NVMeoFClient.pb2.list_subsystems_req()
+            )
+            if subsystems.status != 0:
+                raise DashboardException(
+                    msg=response.error_message,
+                    code=response.status,
+                    http_status_code=NVMeoFClient().NVMeoFError2HTTP.get(response.status, 400),
+                    component="nvmeof",
+                )
+            # Create a new response with initiators information appended
+            for subs in subsystems.subsystems:
+                res = {
+                    'nqn': subs.nqn,
+                    'enable_ha': subs.enable_ha,
+                    'serial_number': subs.serial_number,
+                    'model_number': subs.model_number,
+                    'min_cntlid': subs.min_cntlid,
+                    'max_cntlid': subs.max_cntlid,
+                    'namespace_count': subs.namespace_count,
+                    'max_namespaces': subs.max_namespaces,
+                    'subtype': subs.subtype,
+                    'initiators': [],
+                }
+                # Get list of initiators present on the subsystem
+                hosts = NVMeoFHost().list(subs.nqn)
+                # Decode the bytes object to json
+                json_str = hosts.decode('utf-8')
+                json_obj = json.loads(json_str)
```

Comment:
What type of Python object is this? A list? In that case, instead of converting to/from JSON, you can probably do something like:
```suggestion
                hosts = [host for host in hosts]
```